client_args.go

```diff
@@ -0,0 +1,54 @@
+package client
+
+import (
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/querybuilders"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/transactions"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+)
+
+// RecordTransactionArgs holds the arguments required to record a user transaction.
+type RecordTransactionArgs struct {
+	Metadata    querybuilders.Metadata // Metadata associated with the transaction.
+	Hex         string                 // Hexadecimal string representation of the transaction.
+	ReferenceID string                 // Reference ID for the transaction.
+}
+
+// ParseToRecordTransactionRequest converts RecordTransactionArgs to a RecordTransactionRequest
+// for SPV Wallet API consumption.
+func (r RecordTransactionArgs) ParseToRecordTransactionRequest() *transactions.RecordTransactionRequest {
+	return &transactions.RecordTransactionRequest{
+		Metadata:    r.Metadata,
+		Hex:         r.Hex,
+		ReferenceID: r.ReferenceID,
+	}
+}
+
+// DraftTransactionArgs holds the arguments required to create user draft transaction.
+type DraftTransactionArgs struct {
+	Config   response.TransactionConfig // Configuration for the transaction.
+	Metadata querybuilders.Metadata     // Metadata related to the transaction.
+}
+
+// ParseToDraftTransactionRequest converts DraftTransactionArgs to a DraftTransactionRequest
+// for SPV Wallet API consumption.
+func (d DraftTransactionArgs) ParseToDraftTransactionRequest() *transactions.DraftTransactionRequest {
+	return &transactions.DraftTransactionRequest{
+		Config:   d.Config,
+		Metadata: d.Metadata,
+	}
+}
+
+// UpdateTransactionMetadataArgs holds the arguments required to update a user transaction's metadata.
+type UpdateTransactionMetadataArgs struct {
+	ID       string                 // Unique identifier of the transaction to be updated.
```

Comment:
If this is passed to an API shouldn't it also have 'json' annotations to map the actual json to 
```
{
"id" : "some-id",
"metadata": "some-metadata",
}
```

i think rn it will make capital letters to the json object but please correct me if im wrong